src/gei/Program.cs

```diff
@@ -21,10 +21,14 @@ public static async Task Main(string[] args)
                 .AddCommands()
                 .AddSingleton(Logger)
                 .AddSingleton<EnvironmentVariableProvider>()
+                .AddSingleton<BlobServiceClientFactory>()
+                .AddTransient<IBlobServiceClientFactory>(sp => sp.GetRequiredService<BlobServiceClientFactory>())
```

Comment:
The way you generally register these kind of mappings in most of the DI containers is as follows:
```c#
AddTransient<IBlobServiceClientFactory, BlobServiceClientFactory>()
```
The container will take care of automatically instantiating the service with all of its dependencies for you.

Reply:
Updated in https://github.com/github/gh-gei/pull/221/commits/20124f53f834ac969c213e17bdb8caa505a3fcee